language/evm/move-to-yul/src/events.rs

```diff
@@ -186,6 +186,168 @@ impl EventSignature {
     }
 }
 
+/// Generate event emit functions for send_<name> in async move contracts
+/// example: the function send_foo(actor: address, args) will emit an ethereum event
+/// Foo(actor, message_hash, args) where encoding follows the ethereum standard
+pub(crate) fn define_emit_fun_for_send(
```

Comment:
This logics makes sense to me for now. I was wondering if we should add a sequence number to the event to help perform deduplication on the listener side. Maybe we can revisit the definition once we start to implement the listener code.